app/views/pieces/promotion.js.erb

```diff
@@ -4,3 +4,6 @@ $('[data-piece-id="<%= @piece.id %>"]')
               class: "piece", \
               data: { piece_id: @piece.id, x_coord: @piece.x_position, y_coord: @piece.y_position } %>');
 init_draggables_and_droppables();
+
+<% flash.now[:alert] << @game.check_message(current_user) if @game.check?(!@piece.is_white?) %>
+$('.game-board').html('<%= j (render partial: 'games/board') %>');
```

Comment:
Do we have to put the `init_draggables_and_droppables();` after line 9?

Reply:
Yes because the html is reloaded on line 9. Drags and drops would not be initiated of we put it before.

Reply:
Sorry I was unclear in my question. It is currently on line 6. I was wondering if it should be moved to after line 9 because of the reason you stated?